python/tank/platform/software_launcher.py

```diff
@@ -85,16 +89,28 @@ class SoftwareLauncher(object):
     should only be constructed through the :meth:`create_engine_launcher()`
     factory method.
     """
-    def __init__(self, tk, context, engine_name, env):
+    def __init__(self, tk, context, engine_name, env, versions=None, products=None):
         """
         :param tk: :class:`~sgtk.Sgtk` Toolkit instance
-        :param context: :class:`~sgtk.Context` A context object to
-                        define the context on disk where the engine
-                        is operating
+
+        :param context: :class:`~sgtk.Context` A context object to define the
```

Comment:
You're simply following the established pattern here, but normally types should be documented either using
```
:param your_type variable:
```
or
```
:param variable: your_documentation
:type variable: your_type
```

Although it this case maybe

```
:param context: A :class:`~sgtk.Context` object to define the context on disk 
    where the engine is operating.
```

would be even nicer. I'll leave it up to you.